src/macho.ts

```diff
@@ -8,7 +10,7 @@ export class MachoFile {
     private uuid: string | undefined;
     private header: MachoHeader | undefined;
 
-    constructor(public readonly path: string, private headerOffset = 0) { }
+    constructor(public readonly path: string, private headerOffset: number, private size: number) { }
```

Comment:
Why not make both of these default to zero?

Reply:
Responded in discord. For historical purposes the gist of the answer was:

* 0 is an invalid value for size, causing writeFile to break
* guarding writeFile felt like an anti-pattern
* headerOffset can be 0 but required params can't come after optional params
* swapping headerOffset and size is a bad pattern for a public package